packages/modules/event-bus-local/src/services/event-bus-local.ts

```diff
@@ -21,18 +22,24 @@ eventEmitter.setMaxListeners(Infinity)
 
 // eslint-disable-next-line max-len
 export default class LocalEventBusService extends AbstractEventBusModuleService {
+  #isWorkerMode: boolean = true
   protected readonly logger_?: Logger
   protected readonly eventEmitter_: EventEmitter
   protected groupedEventsMap_: StagingQueueType
 
-  constructor({ logger }: MedusaContainer & InjectedDependencies) {
+  constructor(
+    { logger }: MedusaContainer & InjectedDependencies,
+    moduleOptions = {},
+    moduleDeclaration: InternalModuleDeclaration
+  ) {
     // @ts-ignore
     // eslint-disable-next-line prefer-rest-params
     super(...arguments)
 
     this.logger_ = logger
     this.eventEmitter_ = eventEmitter
     this.groupedEventsMap_ = new Map()
+    this.#isWorkerMode = moduleDeclaration.worker_mode !== "server"
```

Comment:
I think we should also emit events if the mode is `shared` I believe?

Reply:
+1, I think this should be `moduleDeclaration.worker_mode === "worker"` – otherwise you won't be able to process events in a setup where one instance serves both purposes.

Reply:
We emit them, we just don't let subscribers on instances that are `server` only.
@olivermrbl the default one is `shared`, so this should work fine.

Reply:
yes I am not sure I get it too, we just prevent the subscribers from registering themselves if you are in server mode only. if you are in shared mode it will just work as it does today. It prevents smaller project to be required to have both server and worker separately

Reply:
Just reverting back here, the behavior seems correct, I just got tricked by the naming at first glance. Maybe we can say `shouldRegisterSubscribers`?

Reply:
Well, it can be used for any purpose that would need to know if we are in worker mode or not. If someone build his own event bus modules (which will happen) they might also use it for something else. So in the base class the information is are we in worker mode, and then it happens that we use it for our subscribers management